tensorflow/compiler/xla/service/pattern_matcher_test.cc

```diff
@@ -1102,5 +1102,75 @@ TEST_F(PatternMatcherTest, CustomCallMatchers) {
       root, m::CustomCall("test_target", m::Parameter(1), m::Parameter(0))));
 }
 
+TEST_F(PatternMatcherTest, OptionalUnaryOp) {
+  constexpr char kModuleStr[] = R"(
+    HloModule test_module
+
+    ENTRY test {
+      p0 = f32[] parameter(0)
+      cos = cosine(p0)
+      ROOT out = f32[] abs(cos)
+    }
+  )";
+  TF_ASSERT_OK_AND_ASSIGN(auto hlo_module,
+                          ParseAndReturnVerifiedModule(kModuleStr));
+  auto* root = hlo_module->entry_computation()->root_instruction();
+
+  EXPECT_TRUE(Match(
+      root,
+      m::OptionalUnaryOp({HloOpcode::kBitcast, HloOpcode::kCos}, m::Abs())));
+  EXPECT_TRUE(
+      Match(root, m::OptionalUnaryOp({HloOpcode::kBitcast, HloOpcode::kCos},
+                                     m::Abs(m::Cos()))));
+  EXPECT_TRUE(Match(
+      root, m::OptionalUnaryOp({HloOpcode::kCos, HloOpcode::kAbs}, m::Cos())));
+  EXPECT_FALSE(Match(
+      root,
+      m::OptionalUnaryOp({HloOpcode::kCos, HloOpcode::kAbs}, m::Bitcast())));
+  EXPECT_FALSE(Match(
+      root,
+      m::OptionalUnaryOp({HloOpcode::kCos, HloOpcode::kBitcast}, m::Cos())));
+
+  std::string description = absl::StrCat(
+      "an HloInstruction which optionally matches a unary operand with any of "
+      "opcodes {",
+      HloOpcodeString(HloOpcode::kCos), ", ",
+      HloOpcodeString(HloOpcode::kBitcast),
+      "} before matching the given pattern.");
+  std::string explanation = absl::StrCat(
+      "HloInstruction doesn't have opcode ",
+      HloOpcodeString(HloOpcode::kBitcast),
+      "\nin out = f32[] abs(f32[] cos) HloInstruction also doesn't have any of "
```

Comment:
These explanations can get pretty long and hard to read, so my intent is to make this as easy for people to read as possible by employing something that approximates a "plain English sentence".

A more clear description to me would be "and also doesn't have one of the opcodes {...}".